libs/api/repository/src/lib/gn4/gn4-repository.ts

```diff
@@ -230,6 +231,28 @@ export class Gn4Repository implements RecordsRepositoryInterface {
     )
   }
 
+  openRecordForDuplication(
+    uniqueIdentifier: string
+  ): Observable<[CatalogRecord, string, false] | null> {
+    return this.loadRecordAsXml(uniqueIdentifier).pipe(
+      switchMap(async (xml) => {
+        const converter = findConverterForDocument(xml)
+        const record = await converter.readRecord(xml)
+        return [record, converter] as [CatalogRecord, BaseConverter<string>]
+      }),
+      switchMap(async ([record, converter]) => {
+        record.uniqueIdentifier = `TEMP-ID-${Date.now()}`
+        record.title = `${record.title} (Copy)`
+        const xml = await converter.writeRecord(record)
```

Comment:
The original XML document should be provided to `writeRecord` so that only the affected parts of the XML are changed, otherwise an completely new XML doc will be generated from scratch.

```suggestion
        const xml = await converter.writeRecord(record, xml)
```